### PR TITLE
support dns search&options

### PR DESCRIPTION
**Pull request by gao-feng**
Signed-off-by: Gao feng <omarapazanadi@gmail.com>

### CODE REVIEW

**Comment by gao-feng** [comment]
https://github.com/hyperhq/runv/pull/463
https://github.com/hyperhq/hyperstart/pull/284

**Comment by feiskyer** [comment]
LGTM. Please update the runv vendor.

**Comment by gnawux** [comment]
close-open to trigger the re-merge/re-test as #576 has been merged

**Comment by gnawux** [comment]
need update protobuf generated code

```
Generated types from proto updated.
--- hack/../_tmp/types.pb.go	2017-04-02 02:44:17.024529037 +0000
+++ hack/../types/types.pb.go	2017-04-02 02:44:23.569360767 +0000
@@ -997,7 +997,7 @@ func (m *PodStats) GetContainersStats()
 
 type CpuStats struct {
 	Usage       *CpuUsage `protobuf:"bytes,1,opt,name=usage" json:"usage,omitempty"`
-	LoadAverage int32     `protobuf:"varint,2,opt,name=LoadAverage,json=loadAverage,proto3" json:"LoadAverage,omitempty"`
+	LoadAverage int32     `protobuf:"varint,2,opt,name=LoadAverage,proto3" json:"LoadAverage,omitempty"`
 }
 
 func (m *CpuStats) Reset()                    { *m = CpuStats{} }
@@ -2272,9 +2272,9 @@ func (m *UserUser) GetAdditionalGroups()
 }
 
 type Ulimit struct {
-	Name string `protobuf:"bytes,1,opt,name=Name,json=name,proto3" json:"Name,omitempty"`
-	Hard uint64 `protobuf:"varint,2,opt,name=Hard,json=hard,proto3" json:"Hard,omitempty"`
-	Soft uint64 `protobuf:"varint,3,opt,name=Soft,json=soft,proto3" json:"Soft,omitempty"`
+	Name string `protobuf:"bytes,1,opt,name=Name,proto3" json:"Name,omitempty"`
+	Hard uint64 `protobuf:"varint,2,opt,name=Hard,proto3" json:"Hard,omitempty"`
+	Soft uint64 `protobuf:"varint,3,opt,name=Soft,proto3" json:"Soft,omitempty"`
 }
 
 func (m *Ulimit) Reset()                    { *m = Ulimit{} }
@@ -2319,7 +2319,7 @@ type UserContainer struct {
 	User          *UserUser              `protobuf:"bytes,13,opt,name=user" json:"user,omitempty"`
 	Labels        map[string]string      `protobuf:"bytes,14,rep,name=labels" json:"labels,omitempty" protobuf_key:"bytes,1,opt,name=key,proto3" protobuf_val:"bytes,2,opt,name=value,proto3"`
 	Id            string                 `protobuf:"bytes,15,opt,name=id,proto3" json:"id,omitempty"`
-	StopSignal    string                 `protobuf:"bytes,17,opt,name=StopSignal,json=stopSignal,proto3" json:"StopSignal,omitempty"`
+	StopSignal    string                 `protobuf:"bytes,17,opt,name=StopSignal,proto3" json:"StopSignal,omitempty"`
 	Ulimits       []*Ulimit              `protobuf:"bytes,18,rep,name=ulimits" json:"ulimits,omitempty"`
 	LogPath       string                 `protobuf:"bytes,19,opt,name=logPath,proto3" json:"logPath,omitempty"`
 }
@@ -3149,7 +3149,7 @@ func (*InfoRequest) ProtoMessage()
 func (*InfoRequest) Descriptor() ([]byte, []int) { return fileDescriptorTypes, []int{70} }
 
 type InfoResponse struct {
-	ID                 string          `protobuf:"bytes,1,opt,name=ID,json=iD,proto3" json:"ID,omitempty"`
+	ID                 string          `protobuf:"bytes,1,opt,name=ID,proto3" json:"ID,omitempty"`
 	Containers         int32           `protobuf:"varint,2,opt,name=containers,proto3" json:"containers,omitempty"`
 	Images             int32           `protobuf:"varint,3,opt,name=images,proto3" json:"images,omitempty"`
 	Driver             string          `protobuf:"bytes,4,opt,name=driver,proto3" json:"driver,omitempty"`
Generated types from proto is out of date. Please run hack/update-generated-proto.sh
The command "hack/verify-generated-proto.sh" failed and exited with 1 during .
```

**Comment by gnawux** [comment]
retest this please, hykins

File: types/types.proto [hunk 1]
```diff
@@ -489,6 +489,8 @@ message UserPod {
   repeated UserService services              = 14;
   PortmappingWhiteList portmappingWhiteLists = 15;
   repeated PortMapping portmappings          = 16;
+  repeated string dnsOptions		     = 17;
```
Comment (feiskyer) [inline types/types.proto]: nit: indent with another space

**Comment by gnawux** [comment]
retest this please, hykins

**Comment by gnawux** [comment]
retest this please, hykins

File: types/types.proto (hunk 1 above)
Comment (gao-feng) [inline types/types.proto]: https://github.com/gao-feng/hyper-public/blob/310ebcb653f2688efcde5f9e5f44adb73ca19ad3/types/types.proto looks good

Comment (gnawux) [inline types/types.proto]: looks a tab in the inserted line, which looks different in different view...

**Pull request merged**